File: app/src/main/java/com/kksionek/photosyncer/data/Friend.java
package com.kksionek.photosyncer.data;

import androidx.annotation.NonNull;

import io.realm.RealmObject;
import io.realm.annotations.PrimaryKey;

public class Friend extends RealmObject implements Person, Comparable<Person> {
    @PrimaryKey
    private String mId;
    private String mName;
    private String mPhoto;

    private boolean mOld;

    public Friend() {
        mId = null;
        mName = null;
        mPhoto = null;
        mOld = false;
    }

    public Friend(String uid, String name, String photo) {
        mId = uid;
        mName = name;
        mPhoto = photo;
        mOld = false;
    }

    @Override
    public int compareTo(@NonNull Person another) {
        return mName.compareTo(another.getName());
    }

    @Override
    public boolean equals(Object o) {
        return o instanceof Person && mName.equals(((Person) o).getName());
    }

    @NonNull
    public String getName() {
        return mName;
    }

    public String getPhoto() {
        return mPhoto;
    }

    public String getId() {
        return mId;
    }

    public String getFacebookId() {
        return mId;
    }

    public void setOld(boolean old) {
        mOld = old;
    }
}

File: app/src/main/java/com/kksionek/photosyncer/data/Friend.kt
package com.kksionek.photosyncer.data

open class Friend constructor(
    override var id: Int = -1,
    override var name: String = "",
    override var photo: String = "",
    var old: Boolean = false
) : Person, Comparable<Person> {

    override fun compareTo(other: Person): Int {
        return name.compareTo(other.name)
    }

    override fun equals(other: Any?): Boolean {
        return other is Person && name == other.name
    }
}
